Tpetra: "declaration shadows a typedef" warnings emitted by Apple clang 10.0.0

@trilinos/tpetra

## Expectations
Builds with clang that use the `-Wshadow` flag should not emit warnings.

## Current Behavior
Three such warnings are emitted by clang when building `MatrixMarket_Tpetra.hpp`.
```
.../include/MatrixMarket_Tpetra.hpp:6896:56: warning: declaration shadows a typedef in 'Writer<SparseMatrixType>' [-Wshadow]
        typedef typename multivector_type::scalar_type scalar_type;
                                                       ^
.../include/MatrixMarket_Tpetra.hpp:5844:54: note: previous declaration is here
      typedef typename SparseMatrixType::scalar_type scalar_type;
                                                     ^
.../include/MatrixMarket_Tpetra.hpp:7021:56: warning: declaration shadows a typedef in 'Writer<SparseMatrixType>' [-Wshadow]
        typedef typename multivector_type::scalar_type scalar_type;
                                                       ^
.../include/MatrixMarket_Tpetra.hpp:5844:54: note: previous declaration is here
      typedef typename SparseMatrixType::scalar_type scalar_type;
                                                     ^
.../include/MatrixMarket_Tpetra.hpp:976:57: warning: declaration shadows a typedef in 'Reader<SparseMatrixType>' [-Wshadow]
        typedef typename ArrayView<const GO>::size_type size_type;
                                                        ^
.../include/MatrixMarket_Tpetra.hpp:220:49: note: previous declaration is here
      typedef Teuchos::ArrayRCP<int>::size_type size_type;
```

## Environment
I'm building on a Mac using Apple clang 10.0.0.

--- a/packages/tpetra/core/inout/MatrixMarket_Tpetra.hpp
+++ b/packages/tpetra/core/inout/MatrixMarket_Tpetra.hpp
@@ -973,7 +973,6 @@ namespace Tpetra {
         using Teuchos::RCP;
         using Teuchos::rcp;
         typedef global_ordinal_type GO;
-        typedef typename ArrayView<const GO>::size_type size_type;
 
         // Construct the CrsMatrix.
         //
@@ -6893,7 +6892,6 @@ namespace Tpetra {
         using Teuchos::REDUCE_MAX;
         using Teuchos::reduceAll;
         using std::endl;
-        typedef typename multivector_type::scalar_type scalar_type;
         typedef Teuchos::ScalarTraits<scalar_type> STS;
         const char prefix[] = "Tpetra::MatrixMarket::writeDenseHeader: ";
 
@@ -7018,7 +7016,6 @@ namespace Tpetra {
         using Teuchos::TypeNameTraits;
         using Teuchos::wait;
         using std::endl;
-        typedef typename multivector_type::scalar_type scalar_type;
         typedef Teuchos::ScalarTraits<scalar_type> STS;
 
         const Comm<int>& comm = * (X.getMap ()->getComm ());

--- a/packages/tpetra/core/src/Tpetra_CrsMatrix_def.hpp
+++ b/packages/tpetra/core/src/Tpetra_CrsMatrix_def.hpp
@@ -8161,8 +8161,6 @@ namespace Tpetra {
     typedef node_type NT;
     typedef CrsMatrix<Scalar, LO, GO, NT> this_type;
     typedef Vector<int, LO, GO, NT> IntVectorType;
-    const LO LINVALID = Teuchos::OrdinalTraits<LO>::invalid ();
-    const GO GINVALID = Teuchos::OrdinalTraits<GO>::invalid ();
     using Teuchos::as;
 
     const bool debug = ::Tpetra::Details::Behavior::debug ();

--- a/packages/tpetra/core/test/Block/BlockCrsMatrix.cpp
+++ b/packages/tpetra/core/test/Block/BlockCrsMatrix.cpp
@@ -1360,7 +1360,6 @@ namespace {
 
     RCP<const Comm<int> > comm = getDefaultComm ();
     const int myRank = comm->getRank ();
-    const int numProcs = comm->getSize ();
     const GST INVALID = Teuchos::OrdinalTraits<GST>::invalid ();
 
     out << "Creating mesh row Map" << endl;

--- a/packages/tpetra/core/test/MultiVector/MultiVector_get2dView.cpp
+++ b/packages/tpetra/core/test/MultiVector/MultiVector_get2dView.cpp
@@ -83,9 +83,7 @@ namespace { // (anonymous)
     out << "Test Tpetra::MultiVector::get2dView and get2dViewNonConst" << endl;
     Teuchos::OSTab tab1 (out);
 
-    // For later user
     Scalar curVal = STS::zero ();
-    mag_type curMagVal = STM::zero ();
 
     // Create a Map with enough rows that we can view a noncontiguous
     // subset of columns without the ratio of rows to columns being
@@ -114,7 +112,6 @@ namespace { // (anonymous)
     // of 0, because 0 is the default fill value.
     out << "Fill X" << endl;
     curVal = STS::one ();
-    curMagVal = STM::zero (); // for later use
     X.template sync<device_type> ();
     X.template modify<device_type> ();
     for (size_t j = 0; j < numCols; ++j) {
@@ -213,7 +210,6 @@ namespace { // (anonymous)
 
       MV X_copy (X, Teuchos::Copy); // make a "backup" of X
       curVal = STS::one ();
-      curMagVal = STM::one ();
       for (size_t j = 0; j < numColsSubset; ++j) {
         const size_t col = colsSubset[j];
         auto X_col_orig = X.getVectorNonConst (col);

